packages/tools/src/utilities/math/basic/BasicStatsCalculator.ts

```diff
@@ -0,0 +1,40 @@
+//Basic calculator
+import { PointInShape, StatisticValue } from '../../../types';
+
+export default function BasicStatsCalculator(
+  points: PointInShape[]
+): StatisticValue[] {
+  let max = -Infinity;
+  let mean = 0;
+  let sumSquares = 0;
+
+  points.forEach((point) => {
+    const newValue = point.value;
+    if (newValue > max) {
+      max = newValue;
+    }
+
+    sumSquares += newValue ** 2;
+    mean += newValue;
+  });
+
+  mean = mean / points.length;
+
+  let stdDev = 0;
+  let stdDevWithSumSquare = 0;
+
+  points.forEach((point) => {
+    const valueMinusMean = point.value - mean;
+    stdDev += valueMinusMean ** 2;
+  });
+
+  stdDevWithSumSquare = Math.sqrt(sumSquares / points.length - mean ** 2);
+  stdDev = Math.sqrt(stdDev / points.length);
+
```

Comment:
for these repetitive loop you should stick to simmple for loop so that browser can optimize it. Also one for loop is enough